Sources/ChatToys/Utils/WebContext.swift

```diff
@@ -205,39 +204,26 @@ extension Array {
     }
 }
 
-public class MarkdownProcessor {
-    private static var lock = NSLock()
+public actor MarkdownProcessor {
```

Comment:
in order to get the actor benefits you need to be using _instance variables_ of the class -- it seems like [actor guarantees do not protect static vars](https://forums.swift.org/t/does-an-actor-protect-static-properties-and-functions/60970). 

Ez solution here would be to remove all the `static`s and have a shared static `MarkdownProcessor`:

```
public actor MarkdownProcessor {
  public static let shared = MarkdownProcessor()
....etc
```